### PR TITLE
API and R bindings

### PR DESCRIPTION
**Pull request by nalinigans**
This is only an initial attempt at an api and a R wrapper to OmicsDS. Putting this PR out for discussion.

### CODE REVIEW

File: bindings/r/src/omicsds_query.cpp [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    if (found == m_features.end()) {
+      m_features.insert({id, m_features.size()});
+      found = m_features.find(id);
+    }
+    return found->second;
+  }
+
+  bool started = false;
+  uint64_t first_sample_id;
+  std::unordered_map<uint64_t, std::vector<float>> m_feature_scores;
```
[the remark this answers is not in the document]
Comment (nalinigans) [reply bindings/r/src/omicsds_query.cpp:175]: Yes, the backing store is `std::unordered_map`. I was just aiming for performance first, will move to `std::map` if the order is an issue. Hoping @kkavya0391 or @wrighth-omics weigh in here.

File: src/main/cpp/api/omicsds.cc [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+void OmicsDS::query_features(OmicsDSHandle handle, std::vector<std::string>& features, std::array<int64_t, 2>& sample_range, feature_process_fn_t proc) {
+  std::shared_ptr<OmicsExporter> instance;
+  {
+    const std::lock_guard<std::mutex> lock(omicsds_mutex);
+    instance = omicsds_instances.at(handle);
+  }
+  logger.debug("New Query");
+  FeatureProcessor feature_processor(features, sample_range, proc);
+  process_function bound = std::bind(&FeatureProcessor::process, std::ref(feature_processor), std::placeholders::_1,  std::placeholders::_2);
+  std::array<int64_t, 2> position_range = {0, std::numeric_limits<int64_t>::max()};
```
[the remark this answers is not in the document]
Comment (nalinigans) [reply src/main/cpp/api/omicsds.cc:121]: Yes, `feature_range` sounds good as it is a `query_features` function after all. This pass at an api and R bindings is an initial support at `query_features`, so can easily make this change. Thanks.

[the remark this answers is not in the document]
Comment (nalinigans) [reply src/main/cpp/api/omicsds.cc:121]: I will probably call it `encoded_features_range` as it does not really map with the features directly.

**Review by webberlh — APPROVED** [review]
Looks good to me!

**Pull request merged**